report_thesis/src/sections/background/ensemble_learning_models/xgboost.tex

```diff
@@ -1 +1,23 @@
-\subsubsection{XGBoost}
\ No newline at end of file
+\subsubsection{XGBoost}
+We now describe XGBoost based on \citet{ChenGuestrin2016}.
+
+\gls{xgboost} is an advanced implementation of gradient boosting designed for speed and performance. 
+It builds upon traditional boosting techniques by introducing several algorithmic enhancements to improve execution speed and model performance.
+
+The core mechanism of \gls{xgboost} can be summarized in the following way:
+The algorithm first sorts the feature values and visits the data in sorted order to evaluate all possible split by accumulating gradient statistics.
+Then, each possible split is evaluated using a regularized objective function. 
+This function incorporates both a loss function that measures the difference between predicted and actual values and a regularization term that penalizes the complexity of the model.
+This additional regularization term helps prevent overfitting by smoothening the final learned weights.
+For each candidate split, the algorithm calculates a split score based on the accumulated statistics. 
+The algorithm chooses the split with the best split score, which minimizes the regularized objective function.
+This chosen split it used to partition the data into two groups.
+This process is recursively applied to each partition, building a tree structure by continually splitting the data to minimize the regularized objective function at each node.
```

Comment:
This is the only point not addressed by the paper, but since this is a general trait of tree based algorithms I suppose its fine to keep. 
Should I be more explicit about this?